tests/framework/extensions/workloads/podstatus.go

```diff
@@ -45,15 +48,35 @@ func StatusPods(client *rancher.Client, clusterID string, listOpts metav1.ListOp
 
 	var podResults []string
 	var podErrors []error
-	podResults = append(podResults, "pods Status:\n")
 
 	for _, pod := range podList {
 		podStatus := pod.Status.Phase
 		if podStatus == "Succeeded" || podStatus == "Running" {
 			podResults = append(podResults, fmt.Sprintf("INFO: %s: %s\n", pod.Name, podStatus))
 		} else {
-			podErrors = append(podErrors, fmt.Errorf("ERROR: %s: %s", pod.Name, podStatus))
+			podErrors = append(podErrors, fmt.Errorf("ERROR: %s: %s\n", pod.Name, podStatus))
 		}
 	}
 	return podResults, podErrors
 }
+
+// WaitPodTerminated is a helper function that uses wait.Poll() to verify if all pods with podName
+// have terminated correctly or if there is still a pod runnnig.
+func WaitPodTerminated(client *rancher.Client, clusterID string, podName string) bool {
+	err := wait.Poll(5*time.Second, 2*time.Minute, func() (done bool, err error) {
+		podResults, _ := StatusPods(client, clusterID, metav1.ListOptions{})
+		for _, pod := range podResults {
+			p := strings.Split(pod, ": ")
+			if strings.HasPrefix(p[1], podName) && strings.HasPrefix(p[2], "Running") {
```

Comment:
My major concern about this function is that it could be vacuously true. That is to say, if it doesn't find any pod with the given prefix, we are going to succeed and pass the test despite this being broken and not finding any pods. If you have ran a live test and made sure that this is actually finding the pod as expected we can move it forward, but otherwise, it might be good to make sure that len(podResults) > 0 or something similar.